scalafix-core/shared/src/main/scala/scalafix/internal/rule/OptionConstructor.scala

```diff
@@ -0,0 +1,23 @@
+package scalafix.internal.rule
+
+import scala.meta._
+
+import scalafix.{Patch, SemanticRule, SemanticdbIndex}
+import scalafix.rule.{RuleCtx, RuleName}
+
+final case class OptionConstructor(index: SemanticdbIndex)
+    extends SemanticRule(index, RuleName("OptionConstructor")) {
+
+  override def description: String =
+    "Rewrite that replaces Some() constructor by Option() constructor"
+
+  private def isSome(term: Term) =
+    index.symbol(term).contains(Symbol("_root_.scala.Some."))
```

Comment:
I would advise to use a SymbolMatcher here, which defines an `unapply` that can be used in the pattern match below. Here's an example to use as a reference:

https://github.com/typelevel/cats/blob/49374a4137823ec82782b17fac8aafcea1979d96/scalafix/rules/src/main/scala/fix/Cats_v1_0_0.scala#L143-L145

You would then use it as

```scala
  val someConstructor = //... SymbolMatcher
   
  // ...

  case Term.Apply(some(fun), _) => // ... replace fun
```

Reply:
That's way much better.
Little question: Do you know what the dot at the end is? Ex: `_root_.scala.Some.`

Reply:
IIRC It's to distinguish the type from the object (# vs .)